### PR TITLE
Add rollup to build libruary.

### PR DESCRIPTION
**Pull request by OperKH**
Also:
- Add .editorconfig.
- Add .prettierrc to be able quick format w/o brakiong es-lint rules.
- Add .eslintrc.js to get rid of vars.
- Move "./index.js" to "./src/index.js".

To make build just run `npm run build` it will create `./dist` folder with 3 types of modules.

Rollup config checks eslint, transpiles ES6+ code to ES5 and uglifies code.

Without build modules I can't use this lib in my project because libraries should be distributed as ES5 code to not break compatibility with <IE11 and because we are always do not compile `node_modules` code.

Before this PR I saw `const` in my `vendors.js` file.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ function rfdc (opts) {
   function cloneArray (a, fn) {
     const keys = Object.keys(a)
     const a2 = new Array(keys.length)
-    for (var i = 0; i < keys.length; i++) {
+    for (let i = 0; i < keys.length; i++) {
```
Comment (BridgeAR) [inline src/index.js]: It is intentional to use `var` here due to supporting older Node.js versions which did not handle that well.

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "semi": false,
+  "singleQuote": true,
+  "trailingComma": "none"
+}
```
Comment (BridgeAR) [inline .prettierrc]: I don't think this file should be added.

File: .npmignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 .nyc_output
 coverage
-benchmark
\ No newline at end of file
+benchmark
+node_modules
```
Comment (BridgeAR) [inline .npmignore]: The test file, the rollup config and the editorconfig etc. should also be ignored.

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  extends: 'standard',
+  rules: {
+    'no-var': 2
+  }
+}
```
Comment (BridgeAR) [inline .eslintrc.js]: Is this required for the eslint module?

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 2
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (BridgeAR) [inline .editorconfig]: Even though I think it's fine to add this, I would rather have it in a separate PR.

**Comment by davidmarkclements** [comment]
> * Add .editorconfig.
I'm not totally against adding this, but I'm also not that interested in it – But I  agree with @BridgeAR it should be a separate PR 

> * Add .prettierrc to be able quick format w/o brakiong es-lint rules.

This projects uses standard, (see devDependencies) - use `standard --fix` to reformat

> * Add .eslintrc.js to get rid of vars.

why? there's nothing wrong with using vars 

> * Move "./index.js" to "./src/index.js".
> 
> To make build just run `npm run build` it will create `./dist` folder with 3 types of modules.
> 
> Rollup config checks eslint, transpiles ES6+ code to ES5 and uglifies code.
> 
> Without build modules I can't use this lib in my project because libraries should be distributed as ES5 code to not break compatibility with <IE11 and because we are always do not compile `node_modules` code.
> 
> Before this PR I saw `const` in my `vendors.js` file.

I would rather convert all code to ES5 manually than introduce transpilation overhead.

File: .eslintrc.js (hunk 4 above)
Comment (OperKH) [reply .eslintrc.js]: Will remove it.

File: .editorconfig (hunk 5 above)
Comment (OperKH) [reply .editorconfig]: Will remove it. And create separate PR.

File: .prettierrc (hunk 2 above)
Comment (OperKH) [reply .prettierrc]: Will remove it.

**Comment by OperKH** [comment]
Build is more convenient for front-end projects:
1. Code already minified
2. Code 100% supported by IE11 (ES5)
3. It's able to use any type of modules, so it adds support of any bundle system on front-end projects.

**Pull request closed without merging**